src/main.ts

```diff
@@ -100,6 +108,173 @@ export class Minifier {
+  // the information gathered from this step are used to figure out which types are internal to the
+  // scope that the minifier is working with and which are external. This allows the minifier to 
+  // rename properties more correctly.
+  private _preprocessVisit(node: ts.Node) {
+    switch (node.kind) {
+      case ts.SyntaxKind.CallExpression: {
+        var callExpr = <ts.CallExpression>node;
+        var lhsSymbol = this._typeChecker.getSymbolAtLocation(callExpr.expression);
+
+        let paramSymbols: ts.Symbol[] = [];
+
+        // TODO: understand cases of multiple declarations, pick first declaration for now
+        if (!lhsSymbol || !((<any>lhsSymbol.declarations[0]).parameters)) {
+          this._preprocessVisitChildren(node);
+          break;
+        } else {
+          (<any>lhsSymbol.declarations[0])
+              .parameters.forEach((param) => { paramSymbols.push(param.type.symbol); });
+
+          let argsSymbols: ts.Symbol[] = [];
+
+          // right hand side argument has actual type of parameter
+          callExpr.arguments.forEach(
+              (arg) => { argsSymbols.push(this._typeChecker.getTypeAtLocation(arg).symbol); });
+
+          // Casting from: Use site symbol, to: actual parameter type
+          paramSymbols.forEach((sym, i) => { this._recordCast(sym, argsSymbols[i]); });
+
+          this._preprocessVisitChildren(node);
+          break;
+        }
+      }
+      case ts.SyntaxKind.VariableDeclaration: {
+        let varDecl = <ts.VariableDeclaration>node;
+        if (varDecl.initializer && varDecl.type) {
+          let varDeclTypeSymbol = this._typeChecker.getTypeAtLocation(varDecl.type).symbol;
+          let initTypeSymbol = this._typeChecker.getTypeAtLocation(varDecl.initializer).symbol;
+
+          // Casting from: initializer's type symbol, to: actual variable declaration's annotated
+          // type
+          this._recordCast(initTypeSymbol, varDeclTypeSymbol);
+        }
+        this._preprocessVisitChildren(node);
+        break;
+      }
+      case ts.SyntaxKind.ReturnStatement: {
+        // check if there is an expression on the return statement since it's optional
+        if (node.parent.kind !== ts.SyntaxKind.SourceFile &&
+            (<ts.ReturnStatement>node).expression) {
+          let symbolReturn =
+              this._typeChecker.getTypeAtLocation((<ts.ReturnStatement>node).expression).symbol;
+
+          let methodDeclAncestor = this._getAncestor(node, ts.SyntaxKind.MethodDeclaration);
+          let funcDeclAncestor = this._getAncestor(node, ts.SyntaxKind.FunctionDeclaration);
+          let ancestor;
+
+          // early exit if no ancestor that is method or function declaration
+          if (!methodDeclAncestor && !funcDeclAncestor) {
+            this._preprocessVisitChildren(node);
+            break;
+          }
+
+          // if node has method declaration, parent is method declaration
```

Comment:
compress the 10 lines below to `let funcLikeDecl = <ts...> (methodDeclAncestor || funcDeclAncestor)`, also the comments are too obvious.